### ISSUE TITLE
Validation Error in Bioimage Model Zoo

### ISSUE THREAD

**Issue by crtorreg** [bioimage-io/archive.bioimage.io#212]
Hi, 

I am trying to upload the new version of 3D UNet - arabidopsis - ZeroCostDL4Mic (as other new models) and it gives me the following error during the uploading when clicking the validate bottom. 

```
Error: {
  "name": "{name: 3D UNet - arabidopsis - ZeroCostDL4Mic, ...}",
  "source_name": "{name: 3D UNet - arabidopsis - ZeroCostDL4Mic, ...}",
  "error": "cannot pickle 'pyodide.JsProxy' object",
  "warnings": {},
  "traceback": [
    "  File \"/lib/python3.9/site-packages/bioimageio/spec/commands.py\", line 66, in validate\n    raw_rd = load_raw_resource_description(rdf_source, update_to_current_format=update_format)\n",
    "  File \"/lib/python3.9/site-packages/bioimageio/spec/io_.py\", line 232, in load_raw_resource_description\n    raw_rd = schema.load(data)\n",
    "  File \"/lib/python3.9/site-packages/marshmallow/schema.py\", line 719, in load\n    return self._do_load(\n",
    "  File \"/lib/python3.9/site-packages/marshmallow/schema.py\", line 844, in _do_load\n    processed_data = self._invoke_load_processors(\n",
    "  File \"/lib/python3.9/site-packages/marshmallow/schema.py\", line 1090, in _invoke_load_processors\n    data = self._invoke_processors(\n",
    "  File \"/lib/python3.9/site-packages/marshmallow/schema.py\", line 1220, in _invoke_processors\n    data = processor(data, many=many, **kwargs)\n",
    "  File \"/lib/python3.9/site-packages/bioimageio/spec/model/v0_4/schema.py\", line 388, in add_weights_format_key_to_weights_entry_value\n    data = deepcopy(data)  # Schema.validate() calls pre_load methods, thus we should not modify the input data\n",
    "  File \"/lib/python3.9/copy.py\", line 153, in deepcopy\n    y = copier(memo)\n",
    "  File \"<exec>\", line 274, in __deepcopy__\n",
    "  File \"/lib/python3.9/copy.py\", line 146, in deepcopy\n    y = copier(x, memo)\n",
    "  File \"/lib/python3.9/copy.py\", line 230, in _deepcopy_dict\n    y[deepcopy(key, memo)] = deepcopy(value, memo)\n",
    "  File \"/lib/python3.9/copy.py\", line 161, in deepcopy\n    rv = reductor(4)\n"
  ],
  "nested_errors": {}
}
```

**Comment by constantinpape** [comment]
Thanks for reporting this @crtorreg! This looks like an issue on the javascript side to me, so would probably be good if @oeway could have a look at it.
Could you upload the model somewhere for us to test, e.g. Google Drive with public access permissions?

**Comment by deepimagej** [comment]
Yes, for sure. This is the link of the model in google drive: [model](https://drive.google.com/drive/folders/1PWQZG_OnfEgqx1A8KthD_HpeW4nj3DTH?usp=sharing)

Thank you!

**Comment by crtorreg** [comment]
Yes, for sure! This is the link to the model in google drive: [model](https://drive.google.com/drive/folders/1PWQZG_OnfEgqx1A8KthD_HpeW4nj3DTH?usp=sharing)

Thank you!

**Comment by constantinpape** [comment]
I checked and I can  succesfully validate the model locally. @oeway could you check why this model fails the upload validation?

**Comment by oeway** [comment]
@crtorreg @constantinpape I just investigated into the issue, it appears that the problem is coming from the `timestamp`, we expect it to be a string, but in @crtorreg 's model, the timestamp is not explicitly quoted as a string, and when we parse it in js, it try to convert into a Date object, and that cause the issue of validation. (By default, js-yaml parser parses data-strings without quotes as Date objects.)

I also noticed that we were not clear in the spec about the type of `timestamp`, see [issue](https://github.com/bioimage-io/spec-bioimage-io/issues/364).

For now I have added some hack to fix the type conversion to the validator plugin, it should work now. However, it might be better if @crtorreg can also add quotes around the timestamp value.

In the meantime, i will also submit an issue to pyodide to make sure we can convert the js Date object to python.
